runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -81,24 +76,32 @@
 {
 	private static final int PANEL_EXPANDED_WIDTH = PluginPanel.PANEL_WIDTH + PluginPanel.SCROLLBAR_WIDTH;
 	public static final BufferedImage ICON;
+	private static final BufferedImage SIDEBAR_OPEN;
+	private static final BufferedImage SIDEBAR_CLOSE;
 
 	static
 	{
-		BufferedImage icon = null;
+		BufferedImage icon;
+		BufferedImage sidebarOpen;
+		BufferedImage sidebarClose;
 
 		try
 		{
 			synchronized (ImageIO.class)
```

Comment:
While this code will not be running concurrently with itself, and that is guaranteed by the spec, ImageIO uses shared resources which can be used concurrently, hence the monitor.

Reply:
I bet that `ImageIO` deals with synchronization on its own then

Reply:
It does not. See #883

Reply:
Then the correct fix is to wrap `ImageIO` and have custom ImageRead that solves the issue. Having `synchronized (ImageIO.class)` all over the code place is not the right way

Reply:
Problem: ImageIO.class has fields that need synchronization
Solution: synchronize ImageIO.class
I see nothing wrong with this.

Reply:
I bet it all works fine now. But issue is not that it doesn't work. Issue is that code is duplicated all over the place. 

Make `BufferedImageFactory` that creates and caches `BufferedImage`'s for you. there can synchronize. It will look much cleaner. But I guess this is not relevant to this PR.